### PR TITLE
Add support for concatenation in Loop

### PR DESCRIPTION
**Pull request by mvafin**
### Details:
 - *Add support for concatenation in Loop*

### Tickets:
 - *101884*

### CODE REVIEW

**Comment by mvafin** [comment]
One test will fail until #15862 is merged

**Review by eaidova — APPROVED** [review]

File: src/frontends/pytorch/src/op_table.cpp [hunk 1]
```diff
@@ -93,6 +94,7 @@ OP_CONVERTER(translate_roll);
 OP_CONVERTER(translate_rsqrt);
 OP_CONVERTER(translate_rsub);
 OP_CONVERTER(translate_select);
+OP_CONVERTER(translate_set_attr);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op_table.cpp:97]:
```suggestion
```

File: src/core/src/op/loop.cpp [hunk 2]
```diff
@@ -42,11 +42,11 @@ void op::v5::Loop::validate_and_infer_types() {
...
 
     if (m_special_body_ports.current_iteration_input_idx >= 0) {
-        const auto& cur_iter_rank = m_bodies[0]
-                                        ->get_parameters()
-                                        .at(m_special_body_ports.current_iteration_input_idx)
-                                        ->get_partial_shape()
-                                        .rank();
+        // Propagate current_iteration input shape and type
+        auto& iter_param = m_bodies[0]->get_parameters().at(m_special_body_ports.current_iteration_input_idx);
+        iter_param->set_element_type(get_input_element_type(0));
```
Comment (itikhono) [inline src/core/src/op/loop.cpp:47]: why 0? get_input_element_type(0)
I believe it should be  get_input_element_type(m_special_body_ports.current_iteration_input_idx)

**Review by itikhono — APPROVED** [review]
the changes in Loop op look good

**Pull request merged**